website/project/model.py

```diff
@@ -852,15 +855,43 @@ class Node(GuidStoredObject, AddonModelMixin, IdentifierMixin):
     # Primary institution node is attached to
     primary_institution = fields.ForeignField('institution')
 
+    affiliated_institutions = fields.ForeignField('institution', list=True)
+
     def add_primary_institution(self, user, inst):
         if inst.auth(user):
             self.primary_institution = inst
+            self.affiliated_institutions.append(inst)
             self.save()
```

Comment:
We generally try to avoid implicit saves. This allows us to save computation when making multiple updates at a time

```python
node.add_primary_institution(...)
node.do_something_else()
node.save()
```